config/config.go

```diff
@@ -887,12 +888,12 @@ func (c *Config) LoadConfigData(data []byte) error {
 }
 
 // UpdateConfig pulls the config file, updates with the new values in newConfig and saves back to the file
-func (c *Config) UpdateConfig(newConfig map[string]interface{}, pluginName string, pluginType string, operationType string) {
+func (c *Config) UpdateConfig(newConfig map[string]interface{}, pluginName string, pluginType string, operationType string) error {
```

Comment:
nit: Maybe instead of ```UpdateConfig``` it's named something like ```SetConfig``` or something, since this isn't always updating per se, as sometimes it's just setting it to a blank map it seems?